qiskit/primitives/base_estimator.py

```diff
@@ -0,0 +1,223 @@
+        psi12_H23_result = e([0, 0, 0], [0, 1, 2], [theta1, theta2, theta3])
+        print(psi12_H23_result)
+"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Callable, Iterable, Optional, Sequence
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.circuit.parametertable import ParameterView
+from qiskit.exceptions import QiskitError
+from qiskit.quantum_info import SparsePauliOp
+
+from .estimator_result import EstimatorResult
+
+
+class BaseEstimator(ABC):
+    """Estimator base class.
+
+    Base class for Estimator that estimates expectation values of quantum circuits and observables.
+    """
+
+    def __init__(
+        self,
+        circuits: Iterable[QuantumCircuit],
+        observables: Iterable[SparsePauliOp],
+        parameters: Optional[Iterable[Iterable[Parameter]]] = None,
+    ):
+        """
+        Creating an instance of an Estimator, or using one in a `with` context opens a session that
+        holds resources until the instance is `close()` ed or the context is exited.
+
+        Args:
+            circuits: quantum circuits that represents quantum states
+            observables: observables
+            parameters: parameters of quantum circuits.
+                Defaults to `[circ.parameters for circ in circuits]`
+                The indexing is such that `parameters[i, j]` is the j-th formal parameter of
+                `circuits[i]`.
+
+        Raises:
+            QiskitError: for mismatch of circuits and parameters list.
+        """
+        self._circuits = tuple(circuits)
+        self._observables = tuple(observables)
+        if parameters is None:
+            self._parameters = tuple(circ.parameters for circ in self._circuits)
+        else:
+            self._parameters = tuple(ParameterView(par) for par in parameters)
+            if len(self._parameters) != len(self._circuits):
+                raise QiskitError(
+                    f"Different number of parameters ({len(self._parameters)} and "
+                    f"circuits ({len(self._circuits)}"
+                )
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc_info):
+        self.close()
+
+    @abstractmethod
+    def close(self):
+        """Close the session and free resources"""
+        ...
+
+    @property
+    def circuits(self) -> tuple[QuantumCircuit, ...]:
+        """Quantum circuits that represents quantum states.
+
+        Returns:
+            quantum circuits
+        """
+        return self._circuits
+
+    @property
+    def observables(self) -> tuple[SparsePauliOp, ...]:
+        """Observables
+
+        Returns:
+            observables
+        """
+        return self._observables
+
+    @property
+    def parameters(self) -> tuple[ParameterView, ...]:
+        """Parameters of quantum circuits
+
+        Returns:
+            parameters[i, j] is the j-th parameter of the i-th circuit.
```

Comment:
```suggestion
        Returns:
            parameters, where parameters[i][j] is the j-th parameter of the i-th circuit.
```

Reply:
also i'm not sure this description is accurate. The j-th parameter of i-th circuit is defined by the circuits themselves. Here I thought we have an opportunity to reorder them in some other way that we want. Otherwise what's the point of this kwarg?

Reply:
Yes, j-th parameter after remapping